File: JavaScript/leetcode75/src/main/Problem40_MaximumLevelSumOfABinaryTree.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxLevelSum = function (root) {
  // if(root === null) return 0;
  let level = 0,
    maxSum = Number.MIN_VALUE,
    res = 0;
  let queue = [];
  queue.push(root);
  while (queue.length > 0) {
    let size = queue.length;
    let sumAtLevel = 0;
    level++;
    while (size > 0) {
      let currNode = queue.shift();
      sumAtLevel += currNode?.val;
      if (currNode?.left !== null) queue.push(currNode?.left);
      if (currNode?.right !== null) queue.push(currNode?.right);
      size--;
    }
    if (maxSum < sumAtLevel) {
      maxSum = sumAtLevel;
      res = level;
    }
  }
  return res;
};

module.exports = maxLevelSum;

File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem40_MaximumLevelSumOfABinaryTree.java
package org.phoenix.leetcode.challenges;

import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import java.util.LinkedList;
import java.util.Queue;

public class Problem40_MaximumLevelSumOfABinaryTree {
    public int maxLevelSum(TreeNode root) {
        int level = 0;
        int ans = 0, max = Integer.MIN_VALUE;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while (!queue.isEmpty()) {
            int size = queue.size();
            int sum = 0;
            level++;
            while (size > 0) {
                TreeNode curr = queue.poll();
                sum += curr.val;
                if (curr.right != null) queue.offer(curr.right);
                if (curr.left != null) queue.offer(curr.left);
                size--;
            }
            if (max < sum) {
                max = sum;
                ans = level;
            }
        }
        return ans;
    }
}
